### PR TITLE
pythonPackages.progressbar2: Fix checkPhase

### LINKED ISSUE #91492: python37Packages.progressbar2 fails to build with "error: option --addopts not recognized"
**Describe the bug**

When trying to build the package

**To Reproduce**

Any of 
* `nix-env -iA nixos.python37Packages.progressbar2`
* look at https://hydra.nixos.org/build/123012472

**Additional context**

According to https://hydra.nixos.org/api/scmdiff?uri=https%3A%2F%2Fgithub.com%2Fnixos%2Fnixpkgs.git&rev1=bbd12947de648afdf0895f6f816a29955c3b0e38&rev2=6f72f0b0df954e904d9fd1ef8a69fa3cc60059a9&branch=&type=git, `progressbar2` wasn't changed itself, so probably the python framework changed in a way such that you can't give `--addopts` manually anymore.

**Notify maintainers**

@ashgillman 

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.47, NixOS, 20.09pre231796.22a81aa5fc1 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"musnix, nixos-20.09pre231796.22a81aa5fc1, nixos-hardware"`
 - channels(turion): `"home-manager, nixos-19.03-19.03.173691.34c7eb7545d, nixos-unstable-newer-20.09pre231796.22a81aa5fc1, nixpkgs-20.09pre231837.2cd2e7267e5"`
 - nixpkgs: `/home/turion/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - python37Packages.progressbar2
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by turion**
###### Motivation for this change

Fix #91492 .

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by turion** [comment]
@ashgillman

**Comment by turion** [comment]
Unfortunately, this doesn't build for `python2`:

```
$ nix-shell -p nixpkgs-review --run "nixpkgs-review pr 91493"
[...]
3 packages failed to build:
python27Packages.progressbar2 python37Packages.baselines truvari

2 packages built:
python37Packages.progressbar2 python38Packages.progressbar2
```

I don't know how to fix that.

File: pkgs/development/python-modules/progressbar2/default.nix [hunk 1]
```diff
@@ -29,17 +29,17 @@ buildPythonPackage rec {
     pytest sphinx flake8 pytestpep8 pytest-flakes pytestcov
     pytestcache freezegun
   ];
-  # ignore tests on the nix wrapped setup.py and don't flake .eggs directory
+  # ignore tests on the nix wrapped setup.py
   checkPhase = ''
     runHook preCheck
-    ${python.interpreter} setup.py test --addopts "--ignore=.eggs"
+    ${python.interpreter} setup.py test
```
Comment (jonringer) [inline pkgs/development/python-modules/progressbar2/default.nix:35]: they didn't clean cache files before uploading package, that's why some tests are failing
```suggestion
    find . -name '*.py[co]' -delete -o -name __pycache__ -delete
    ${python.interpreter} setup.py test
```

Comment (jonringer) [reply pkgs/development/python-modules/progressbar2/default.nix:35]: made a PR https://github.com/WoLpH/python-progressbar/pull/231

**Comment by jonringer** [comment]
Upstream has released a new patch version with a clean sdist, bumping to the new version should fix this.

**Comment by turion** [comment]
@jonringer Thanks! I bumped the version, let's see if it works :)

/marvin opt-in

**Comment by turion** [comment]
`nixpkgs-review` still fails because `pythonPackages.truvari` still depends on an older version. See https://github.com/spiralgenetics/truvari/pull/52

**Comment by turion** [comment]
> `nixpkgs-review` still fails because `truvari` still depends on an older version. See [spiralgenetics/truvari#52](https://github.com/spiralgenetics/truvari/pull/52). Also, `python37Packages.baselines` fails, but because of an unrelated different dependency.

Both packages were failing before already.

**Comment by turion** [comment]
@jonringer can you review again?

**Pull request merged**